docs/design/lcevc-integration.md

```diff
@@ -0,0 +1,266 @@
+
+# Shaka Player LCEVC integration
+
+# Introduction
+
+This article describes the V-Nova LCEVC Shaka Player integration.
+
+Shaka Player is an open-source JavaScript library for adaptive media. It plays adaptive media formats (such as [DASH](http://dashif.org/) and [HLS](https://developer.apple.com/streaming/)) in a browser, without using plugins or Flash. Instead, Shaka Player uses the open web standards [MediaSource Extensions](https://www.w3.org/TR/media-source/) and [Encrypted Media Extensions](https://www.w3.org/TR/encrypted-media/).
+
+The official project GitHub repository is <https://github.com/google/shaka-player>. This integration is a fork from the Shaka Player open source project from version 3.3.0-pre with commit hash: 73b430248ba76038b466da902b776ecc920b2a35.
+
+# LCEVC integration
+
+## Adding V-Nova required files
+
+The Shaka Player project uses a custom tool from Google called Closure Compiler that is a powerful, low-level JavaScript library designed for building complex and scalable web applications. It is used by many Google web applications, such as Google Search, Gmail, Google Docs, Google+, Google Maps, and others.
+
+It parses the JavaScript code, analyzes it, removes dead code and rewrites and minimizes what's left. It also checks syntax, variable references, and types, and warns about common JavaScript pitfalls.
+
+### Importing Dil.js
+
+In order to import the necessary V-Nova libraries we followed the approach that other externals libraries are using. The necessary V-Nova files need to be imported in the HTML page that is going to use the Shaka Player library.
+
+We have at as an npm package : <https://www.npmjs.com/package/lcevc_dil.js>
+
+```javascript
+  <!-- MPEG-5 Part2 LCEVC support is enabled by including this: -->
+  <script defer src="../node_modules/lcevc_dil.js/dist/lcevc_dil.min.js"></script>
+```
+
+To allow the Closure compiler to use the objects and methods that are exported by the DIL.js we created an `extern`.
+
+### Defining an Extern for LCEVC
+
+Externs are declarations that tell Closure Compiler the names of symbols that should not be renamed during advanced compilation. They are called externs because these symbols are most often defined by code outside the compilation, such a native code, or third-party libraries. For this reason, externs often also have type annotations, so that Closure Compiler can type check your use of those symbols.
+
+In general, it is best to think of externs as an API contract between the implementor and the consumers of some piece of compiled code. The externs define what the implementor promises to supply, and what the consumers can depend on using. Both sides need a copy of the contract. Externs are similar to header files in other languages.
```

Comment:
I'm not certain this needs to be explained in context of the shaka-player project.

Reply:
These lines are deleted.